<launch>
  <arg name="model" default="$(find mobile_robot)/urdf/13-diffdrive.urdf.xacro"/>
  <arg name="rvizconfig" default="$(find mobile_robot)/rviz/urdf.rviz" />

  <include file="$(find mobile_robot)/launch/gazebo.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" />

  <rosparam command="load"
            file="$(find mobile_robot)/config/joints.yaml"
            ns="mobile_robot_joint_state_controller" />
  <!-- <rosparam command="load"
            file="$(find mobile_robot)/config/head.yaml"
            ns="r2d2_head_controller" />
  <rosparam command="load"
            file="$(find mobile_robot)/config/gripper.yaml"
            ns="r2d2_gripper_controller" /> -->
  <rosparam command="load"
            file="$(find mobile_robot)/config/diffdrive.yaml"
            ns="mobile_robot_diff_drive_controller" />

  <node name="mobile_robot_controller_spawner" pkg="controller_manager" type="spawner"
    args="mobile_robot_joint_state_controller
          mobile_robot_diff_drive_controller
          --shutdown-timeout 3"/>

  <!-- <node name="rqt_robot_steering" pkg="rqt_robot_steering" type="rqt_robot_steering"> -->
    <!-- <param name="default_topic" value="/r2d2_diff_drive_controller/cmd_vel"/> -->
  <!-- </node> -->
</launch>
